crates/forge-runner/src/running.rs

```diff
@@ -275,19 +279,58 @@ pub fn run_test_case(
         None
     };
 
-    let run_result = runner.run_function(
+    let mut vm = VirtualMachine::new(true);
+    let res = runner.run_function_with_vm(
         func,
+        &mut vm,
         &mut test_execution_syscall_handler,
         hints_dict,
         instructions,
         builtins,
     );
 
+    test_execution_syscall_handler
+        .child
+        .child
+        .child
+        .resources
+        .vm_resources += &test_execution_syscall_handler
+        .child
+        .child
+        .cheatnet_state
+        .vm_resources_used
+        .vm_resources;
+    test_execution_syscall_handler
+        .child
+        .child
+        .child
+        .resources
+        .syscall_counter
+        .extend(
+            &test_execution_syscall_handler
+                .child
+                .child
+                .cheatnet_state
+                .vm_resources_used
+                .syscall_counter,
+        );
```

Comment:
What's the logic behind this?

Reply:
We add resources used by called contracts to the resources used in the test code 
and then compute gas on top of it